### PR TITLE
Refactor `EmbodiedAgent` and modify tests

### PR DESCRIPTION
**Pull request by dandansamax**
## Description

Describe your changes in detail.

## Motivation and Context

#144 #143 

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/agents/embodied_agent.py [hunk 1]
```diff
@@ -169,20 +100,30 @@ def step(
...
         # NOTE: Only single output messages are supported
         output_message = output_messages[0]
-        explanation, code = self.get_explanation_and_code(
-            output_message.content)
-        if self.verbose:
-            print_text_animated(self.logger_color +
-                                f"> Explanation:\n{explanation}")
-            print_text_animated(self.logger_color + f"> Code:\n{code}")
+        explanations_list, codes_list =\
+            output_message.extract_text_and_code_prompts()
```
Comment (lightaime) [inline camel/agents/embodied_agent.py:104]: Prefer to use parenthesis for formatting and simplify the naming  
```suggestion
        explanations, codes = (
            output_message.extract_text_and_code_prompts())
```

File: camel/agents/embodied_agent.py [hunk 2]
```diff
@@ -169,20 +100,30 @@ def step(
...
+
+        if codes_list is not None:
+            output_message.content = "\n> Executed Results:"
+            for code in codes_list:
+                global_vars = {
+                    action.name: action
+                    for action in self.action_space
+                }
+                executed_results = code.execute(global_vars)
+                output_message.content += (f"\n{executed_results}")
```
Comment (lightaime) [inline camel/agents/embodied_agent.py:124]: We can improving the output format a bit:
```suggestion
                executed_outputs = code.execute(global_vars)
                output_message.content += (f"- Python standard output:\n{executed_outputs[0])}\n"
                    f"- Local variables:\n{executed_outputs[1]}\n")
                output_message.content += "*" * 50 + "\n"
```

**Review by lightaime — APPROVED** [review]
Awesome! Thanks, @dandansamax. Left some very minor comments

File: camel/agents/embodied_agent.py (hunk 2 above)
Comment (dandansamax) [reply camel/agents/embodied_agent.py:124]: Thank you for the improvements. Let me merge it.

**Comment by dandansamax** [comment]
I forgot to merge the comments. SO sorry.

**Pull request merged**